api/src/main/java/org/pmiops/workbench/exfiltration/ObjectNameLengthServiceImpl.java

```diff
@@ -0,0 +1,207 @@
+  private final BucketAuditQueryService bucketAuditQueryService;
+  private final IamService iamService;
+  private final UserService userService;
+  private final EgressEventDao egressEventDao;
+  private final EgressRemediationService egressRemediationService;
+
+  public ObjectNameLengthServiceImpl(
+      FireCloudService fireCloudService,
+      WorkspaceService workspaceService,
+      BucketAuditQueryService bucketAuditQueryService,
+      IamService iamService,
+      UserService userService,
+      EgressEventDao egressEventDao,
+      @Qualifier("internal-remediation-service")
+          EgressRemediationService egressRemediationService) {
+    this.fireCloudService = fireCloudService;
+    this.workspaceService = workspaceService;
+    this.bucketAuditQueryService = bucketAuditQueryService;
+    this.iamService = iamService;
+    this.userService = userService;
+    this.egressEventDao = egressEventDao;
+    this.egressRemediationService = egressRemediationService;
+  }
+
+  /**
+   * Get file access info from BQ grouped by the pet account, google project and bucket. If the
+   * lengths of the filenames is greater than the threshold, an alert is triggered for the user who
+   * created the files.
+   *
+   * <p>Depending on the policy, the {@link EgressRemediationService} takes care of disabling the
+   * user, emailing him about the problem, and creating the jira ticket. By disabled the user we
+   * make sure that the user doesn't get multiple alerts for the same problem because this method
+   * filters only the active users.
+   */
+  @Override
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public void calculateObjectNameLength() {
+
+    // Call BQ to get the files created by each owner in the past 24 hours. This will only return
+    // the entries with file lengths > THRESHOLD
+    List<BucketAuditEntry> fileAccessInfoFromBQ = getFileAccessInfoFromBQ();
+
+    // Get the workspaces for these entries. To only find relevant ones.
+    Map<String, DbWorkspace> projectIdToDbWorkspace =
+        workspaceService.getWorkspacesByGoogleProject(
+            fileAccessInfoFromBQ.stream()
+                .map(BucketAuditEntry::getGoogleProjectId)
+                .collect(Collectors.toSet()));
+
+    for (BucketAuditEntry bucketAuditEntry : fileAccessInfoFromBQ) {
+      // If the returned google project ID from BQ doesn't exist in the DB, do nothing. This may
+      // happen in a shared env environment such as test. I am not sure if it would happen in other
+      // envs.
+      if (!projectIdToDbWorkspace.containsKey(bucketAuditEntry.getGoogleProjectId())) {
+        continue;
+      }
+
+      DbWorkspace workspace = projectIdToDbWorkspace.get(bucketAuditEntry.getGoogleProjectId());
+      LOGGER.info(
+          String.format(
+              "Found an audit entry that exceeds the threshold, workspace ID: %s, google ID: %s",
+              workspace.getWorkspaceId(), workspace.getGoogleProject()));
```

Comment:
Workspace namespace is more useful then workspace id, and we usually use it as unique identifier of a workspace.